packages/@aws-cdk/core/lib/stack-synthesizers/default-synthesizer.ts

```diff
@@ -19,7 +19,7 @@ export const BOOTSTRAP_QUALIFIER_CONTEXT = '@aws-cdk/core:bootstrapQualifier';
 /**
  * The minimum bootstrap stack version required by this app.
  */
-const MIN_BOOTSTRAP_STACK_VERSION = 4;
+const MIN_BOOTSTRAP_STACK_VERSION = 6;
```

Comment:
Because version 6 (see the bootstrap template) was when this functionality was introduced. I believe the changes made for bootstrap version 5 didn't impact the default synth process, which is why it wasn't bumped; however, I'm mostly pulling that from memory so I could be wrong.